### ISSUE TITLE
c0ban functional test list 

### ISSUE THREAD

**Issue by shohu** [shohu/c0ban#10]
# How to test

See. https://github.com/shohu/c0ban/tree/v0.15.1-new/test

## BASE_SCRIPTS (62/63)

| No | test file | status | note |
|:---|:---:|---:|---:|
|1 |wallet-hd.py |OK |https://github.com/shohu/c0ban/issues/13#issuecomment-372507689 |
|2 |~~wallet-hd.py~~ | | |
|3 |walletbackup.py |OK | https://github.com/shohu/c0ban/issues/13#issuecomment-372509789|
|4 |p2p-fullblocktest.py |OK | Changed "MAX_PROTOCOL_MESSAGE_LENGTH * 4". c0ban block size is 4 times the bitcoin block size <br>https://github.com/shohu/c0ban/issues/19#issuecomment-374518467|
|5 |fundrawtransaction.py |OK | https://github.com/shohu/c0ban/issues/29|
|6 |p2p-compactblocks.py |OK | GetBlockSubsidy<br>OK Speed Changed <br>https://github.com/shohu/c0ban/issues/20|
|7 |segwit.py |OK | https://github.com/shohu/c0ban/issues/21|
|8 |wallet.py |OK | https://github.com/shohu/c0ban/issues/13#issuecomment-372513288|
|9 |wallet-accounts.py |OK | https://github.com/shohu/c0ban/issues/23|
|10 |p2p-segwit.py |OK | https://github.com/shohu/c0ban/issues/22|
|11 |wallet-dump.py |OK | https://github.com/shohu/c0ban/issues/16#issuecomment-373584795|
|12 |listtransactions.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373219728|
|13 |sendheaders.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371714944|
|14 |zapwallettxes.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371385723|
|15 |importmulti.py |OK |https://github.com/shohu/c0ban/issues/11#issuecomment-371385874 |
|16 |mempool_limit.py |OK |https://github.com/shohu/c0ban/issues/11#issuecomment-371385980 |
|17 |merkle_blocks.py |OK | https://github.com/shohu/c0ban/issues/13#issuecomment-372545457|
|18 |receivedby.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373220769 |
|19 |abandonconflict.py |OK | https://github.com/shohu/c0ban/issues/14#issuecomment-372866995|
|20 |bip68-112-113-p2p.py |PEN | c0ban is equal nPowTargetSpacing to SEQUENCE_LOCKTIME_GRANULARITY. So this test has no meaning<br>https://github.com/shohu/c0ban/issues/24|
|21 |rawtransactions.py |OK | https://github.com/shohu/c0ban/issues/13#issuecomment-372546642|
|22 |reindex.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371386399|
|23 |keypool-topup.py |OK |https://github.com/shohu/c0ban/issues/11#issuecomment-371386514 |
|24 |zmq_test.py |OK | https://github.com/shohu/c0ban/issues/14#issuecomment-372880206|
|25 |bitcoin_cli.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371714435|
|26 |mempool_resurrect_test.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373259106|
|27 |~~txn_doublespend.py~~ | | duplicate |
|28 |~~txn_clone.py~~ | | duplicate|
|29 |getchaintips.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373226647|
|30 |rest.py |OK | https://github.com/shohu/c0ban/issues/13#issuecomment-372547054|
|31 |mempool_spendcoinbase.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373232020|
|32 |mempool_reorg.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373232200|
|33 |mempool_persist.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373221926|
|34 |multiwallet.py |OK | https://github.com/shohu/c0ban/issues/13#issuecomment-372547374|
|35 |httpbasics.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371387261|
|36 |multi_rpc.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373276359|
|37 |proxy_test.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371387414|
|38 |signrawtransactions.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371387494|
|39 |disconnect_ban.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371387582|
|40 |decodescript.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371387689|
|41 |blockchain.py |OK | https://github.com/shohu/c0ban/issues/25|
|42 |disablewallet.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371387823|
|43 |net.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371387903|
|44 |keypool.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371387998|
|45 |p2p-mempool.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371695805|
|46 |prioritise_transaction.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371388365|
|47 |invalidblockrequest.py |OK | GetBlockSubsidy <br> OK Speed Change<br> https://github.com/shohu/c0ban/issues/26|
|48 |invalidtxrequest.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371696267|
|49 |p2p-versionbits-warning.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371695577|
|50 |preciousblock.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371388747|
|51 |importprunedfunds.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371388850|
|52 |signmessages.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371389049|
|53 |nulldummy.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371389117|
|54 |import-rescan.py |OK | https://github.com/shohu/c0ban/issues/14#issuecomment-372897689|
|55 |mining.py |OK | https://github.com/shohu/c0ban/issues/15#issuecomment-373222168|
|56 |bumpfee.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371389352|
|57 |rpcnamedargs.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371389441|
|58 |listsinceblock.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371389524|
|59 |p2p-leaktests.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371695421|
|60 |wallet-encryption.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371389681|
|61 |bipdersig-p2p.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371695247|
|62 |bip65-cltv-p2p.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371695072|
|63 |uptime.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371390319|
|64 |resendwallettransactions.py |OK | https://github.com/shohu/c0ban/issues/14#issuecomment-372897736|
|65 |minchainwork.py |OK | https://github.com/shohu/c0ban/issues/11#issuecomment-371438382|
|66 |p2p-acceptblock.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371694932|

## EXTENDED_SCRIPTS (17/18)

| No | test file | status | note |
|:---|:---:|---:|---:|
|1 |pruning.py |PEN | I wait long time, test is not finished... so I put this test on hold<br>Warning! This test requires 4GB of disk space and takes over 30 mins (up to 2 hours)<br>https://github.com/shohu/c0ban/issues/30 |
|2 |smartfees.py |OK |https://github.com/shohu/c0ban/issues/15#issuecomment-373226251|
|3 |maxuploadtarget.py |OK|https://github.com/shohu/c0ban/issues/16#issuecomment-373586694<br>https://github.com/shohu/c0ban/commit/60fac8f03cafae8dc1e064c49c36d37b1ea6cb00|
|4 |mempool_packages.py |OK |https://github.com/shohu/c0ban/issues/15#issuecomment-373226442|
|5 |dbcrash.py |OK |GetBlockSubsidy<br>OK Speed Change<br>https://github.com/shohu/c0ban/issues/27|
|6 |bip68-sequence.py |OK |https://github.com/shohu/c0ban/issues/14#issuecomment-372914927|
|7 |getblocktemplate_longpoll.py |OK |https://github.com/shohu/c0ban/issues/14#issuecomment-372915310 |
|8 |p2p-timeouts.py |OK | https://github.com/shohu/c0ban/issues/12#issuecomment-371694380|
|9 |bip9-softforks.py |OK |https://github.com/shohu/c0ban/issues/12#issuecomment-371683565|
|10 |p2p-feefilter.py |OK |https://github.com/shohu/c0ban/issues/14#issuecomment-372915498 |
|11 |rpcbind_test.py |PEN |https://github.com/shohu/c0ban/issues/11#issuecomment-371345996 |
|12 |assumevalid.py |OK | GetBlockSubsidy <br> OK Speed Change<br>https://github.com/shohu/c0ban/issues/32|
|13 |example_test.py |OK |https://github.com/shohu/c0ban/issues/12#issuecomment-371693236 |
|14 |txn_doublespend.py |OK |https://github.com/shohu/c0ban/issues/28|
|15 |txn_clone.py |OK |https://github.com/shohu/c0ban/issues/14#issuecomment-372916224 |
|16 |forknotify.py |OK |https://github.com/shohu/c0ban/issues/11#issuecomment-371340785 |
|17 |invalidateblock.py |OK |https://github.com/shohu/c0ban/issues/11#issuecomment-371340616|
|18 |replace-by-fee.py |OK | |

**Comment by shohu** [comment]
* ref. test_runner.py 
```
BASE_SCRIPTS= [
    # Scripts that are run by the travis build process.
    # Longest test should go first, to favor running tests in parallel
    'wallet-hd.py',
    'walletbackup.py',
    # vv Tests less than 5m vv
    'p2p-fullblocktest.py',
    'fundrawtransaction.py',
    'p2p-compactblocks.py',
    'segwit.py',
    # vv Tests less than 2m vv
    'wallet.py',
    'wallet-accounts.py',
    'p2p-segwit.py',
    'wallet-dump.py',
    'listtransactions.py',
    # vv Tests less than 60s vv
    'sendheaders.py',
    'zapwallettxes.py',
    'importmulti.py',
    'mempool_limit.py',
    'merkle_blocks.py',
    'receivedby.py',
    'abandonconflict.py',
    'bip68-112-113-p2p.py',
    'rawtransactions.py',
    'reindex.py',
    # vv Tests less than 30s vv
    'keypool-topup.py',
    'zmq_test.py',
    'bitcoin_cli.py',
    'mempool_resurrect_test.py',
    'txn_doublespend.py --mineblock',
    'txn_clone.py',
    'getchaintips.py',
    'rest.py',
    'mempool_spendcoinbase.py',
    'mempool_reorg.py',
    'mempool_persist.py',
    'multiwallet.py',
    'httpbasics.py',
    'multi_rpc.py',
    'proxy_test.py',
    'signrawtransactions.py',
    'disconnect_ban.py',
    'decodescript.py',
    'blockchain.py',
    'disablewallet.py',
    'net.py',
    'keypool.py',
    'p2p-mempool.py',
    'prioritise_transaction.py',
    'invalidblockrequest.py',
    'invalidtxrequest.py',
    'p2p-versionbits-warning.py',
    'preciousblock.py',
    'importprunedfunds.py',
    'signmessages.py',
    'nulldummy.py',
    'import-rescan.py',
    'mining.py',
    'bumpfee.py',
    'rpcnamedargs.py',
    'listsinceblock.py',
    'p2p-leaktests.py',
    'wallet-encryption.py',
    'bipdersig-p2p.py',
    'bip65-cltv-p2p.py',
    'uptime.py',
    'resendwallettransactions.py',
    'minchainwork.py',
    'p2p-acceptblock.py',
]

EXTENDED_SCRIPTS = [
    # These tests are not run by the travis build process.
    # Longest test should go first, to favor running tests in parallel
    'pruning.py',
    # vv Tests less than 20m vv
    'smartfees.py',
    # vv Tests less than 5m vv
    'maxuploadtarget.py',
    'mempool_packages.py',
    'dbcrash.py',
    # vv Tests less than 2m vv
    'bip68-sequence.py',
    'getblocktemplate_longpoll.py',
    'p2p-timeouts.py',
    # vv Tests less than 60s vv
    'bip9-softforks.py',
    'p2p-feefilter.py',
    'rpcbind_test.py',
    # vv Tests less than 30s vv
    'assumevalid.py',
    'example_test.py',
    'txn_doublespend.py',
    'txn_clone.py --mineblock',
    'forknotify.py',
    'invalidateblock.py',
    'replace-by-fee.py',
]
```
